src/renderer/hooks/store.js

```diff
@@ -86,11 +107,26 @@ export const useExperimentsFlagsStore = createPersistedStore(
   experimentsFlagsStoreSlice,
   'experiments-flags'
 )
-export const useBackgroundMapStore = createPersistedStore(
+
+const useBackgroundMapState = createPersistedStore(
   backgroundMapStoreSlice,
   'background-maps'
 )
 export const usePersistedUiStore = createPersistedStore(
   persistedUiStoreSlice,
   'ui'
 )
+
+export const useBackgroundMapStore = () => {
+  const backgroundMapsEnabled = useExperimentsFlagsStore(
+    store => store.backgroundMaps
+  )
+
+  const [mapStyle, setMapStyle] = useBackgroundMapState(store =>
+    backgroundMapsEnabled
+      ? [store.mapStyleMapServer, store.setMapStyleServer]
+      : [store.mapStyleLegacy, store.setMapStyleLegacy]
+  )
+
+  return { mapStyle, setMapStyle }
```

Comment:
ideally this would return an array, but I am not too sure how to the equivalent of `as const`

Reply:
I figured out how to return an array
<img width="629" alt="image" src="https://github.com/digidem/mapeo-desktop/assets/67773827/2f02e0b4-6fd0-4a93-9e02-cfdf47bbde80">